src/2048-expert/index.js

```diff
@@ -223,18 +229,38 @@ function checkWin() {
   }
 }
 
-// a loss happens when all cells are full
+// a loss happens when all cells are full and they are not mergeable 
 function checkLost() {
+  if (checkisFull()) {
+    // check if any tiles are mergeable
+    for (let i = 0; i < totalCell-6; i++) {
+      // check vertical
```

Comment:
Can we reuse the `mergeHorizontal()` and `mergeVertical()` functions here? Unsure...

Reply:
After rereading the issue, this comment isn't relevant as we don't want to attempt to merge the blocks at this point.